drivers/storage/portworx/deployment.go

```diff
@@ -359,6 +396,32 @@ func (p *portworx) GetStoragePodSpec(
 	return podSpec, nil
 }
 
+func configureStorageNodeSpec(node *corev1alpha1.StorageNode, config *cloudstorage.Config) {
+	node.Spec = corev1alpha1.StorageNodeSpec{CloudConfig: []corev1alpha1.StorageNodeCloudDriveConfig{}}
+	for _, conf := range config.CloudStorage {
+		sc := corev1alpha1.StorageNodeCloudDriveConfig{
+			Type:      conf.Type,
+			IOPS:      conf.IOPS,
+			SizeInGiB: conf.SizeInGiB,
+			Options:   conf.Options,
+		}
+		node.Spec.CloudConfig = append(node.Spec.CloudConfig, sc)
+	}
+	node.Spec.StorageInstancesPerZone = config.StorageInstancesPerZone
+}
+
+func (p *portworx) getStorageNodes() (*corev1alpha1.StorageNodeList, error) {
+	nodeList := &corev1alpha1.StorageNodeList{}
+	err := p.k8sClient.List(context.TODO(), &client.ListOptions{}, nodeList)
```

Comment:
In the list options we should filter the storage nodes by the cluster's namespace and label selector from `p.GetSelectorLabels()`. Also return only those StorageNodes that are owned by the current StorageCluster..